molgenis-data/src/main/java/org/molgenis/data/support/DefaultEntityMetaData.java

```diff
@@ -77,15 +77,10 @@ public DefaultEntityMetaData(EntityMetaData entityMetaData)
 		this.extends_ = extends_ != null ? new DefaultEntityMetaData(extends_) : null;
 		this.backend = entityMetaData.getBackend();
 
-		// deep copy attributes
-		// TODO: Fails dramatically for self-referencing entities.
 		Iterable<AttributeMetaData> attributes = entityMetaData.getAttributes();
 		if (attributes != null)
 		{
-			for (AttributeMetaData attributeMetaData : attributes)
-			{
-				addAttributeMetaData(new DefaultAttributeMetaData(attributeMetaData));
-			}
+			addAllAttributeMetaData(attributes);
```

Comment:
So now if an entity has an attribute that refers to itself, what happens if you copy it? The copied attribute refers to the old version, I think?